src/mctpd.c

```diff
@@ -1666,6 +1708,161 @@ static int message_read_hwaddr(sd_bus_message *call, dest_phys* dest)
 	return 0;
 }
 
+static int setup_static_eid(ctx *ctx)
+{
+	char **object_paths;
+	int i = 0, net;
+	dest_phys desti = {0}, *dest = &desti;
+	peer *peer = NULL;
+	struct sockaddr_mctp_ext addr;
+	struct mctp_ctrl_cmd_get_eid req = {0};
+	struct mctp_ctrl_resp_get_eid *resp = NULL;
+	req.ctrl_hdr.rq_dgram_inst = RQDI_REQ;
+	req.ctrl_hdr.command_code = MCTP_CTRL_CMD_GET_ENDPOINT_ID;
+	sd_bus_error error = SD_BUS_ERROR_NULL;
+	sd_bus_message *message = NULL;
+
+	int rc = sd_bus_call_method(ctx->bus,
+		OPENBMC_OBJECT_MAPPER, OPENBMC_OBJECT_MAPPER_OBJECT_PATH,
+		OPENBMC_OBJECT_MAPPER_IFACE, "GetSubTreePaths",
+		&error, &message, "sias", "/xyz/openbmc_project/", 0, 1,
+		OPENBMC_MCTP_CONFIG_IFACE);
+	if (rc < 0) {
+		warnx("Failed to issue method call: %s\n", error.message);
+		goto out;
+	}
+
+	sd_bus_message_read_strv(message, &object_paths);
+	message = sd_bus_message_unref(message);
+
+	// Check if there is configuration found
+	if (!object_paths)
+		goto out;
+
+	while(object_paths[i] != NULL) {
+		rc = sd_bus_get_property(ctx->bus,
+			OPENBMC_ENTITY_MANAGER, object_paths[i],
+			OPENBMC_MCTP_CONFIG_IFACE, "EndpointId",
+			&error, &message, "t");
+		if (rc < 0) {
+			warnx("Failed to get property: %s\n", error.message);
+			goto out;
+		}
+		uint64_t data = 0;
+		sd_bus_message_read(message, "t", &data);
+		mctp_eid_t eid = (mctp_eid_t)data;
+
+		peer = find_peer_by_addr(ctx, eid, net);
+		if (peer) {
+			// Endpoint already added
+			i++;
+			continue;
+		}
+
+		message = sd_bus_message_unref(message);
+
+		rc = sd_bus_get_property(ctx->bus,
+			OPENBMC_ENTITY_MANAGER, object_paths[i],
+			OPENBMC_MCTP_CONFIG_IFACE, "Address",
+			&error, &message, "t");
+		if (rc < 0) {
+			warnx("Failed to get property: %s\n", error.message);
+			goto out;
+		}
+
+		sd_bus_message_read(message, "t", &data);
+
+		size_t size = sizeof(uint8_t);
+		memcpy(dest->hwaddr, &data, size);
+		dest->hwaddr_len = size;
+
+		message = sd_bus_message_unref(message);
+
+		rc = sd_bus_get_property(ctx->bus,
+			OPENBMC_ENTITY_MANAGER, object_paths[i],
+			OPENBMC_MCTP_CONFIG_IFACE, "Bus",
+			&error, &message, "t");
+		if (rc < 0) {
+			warnx("Failed to get property: %s", error.message);
+			goto out;
```

Comment:
`break` ?

Reply:
You're also generating the same error message as above, so there will be no way to tell which path failed.

Reply:
Done. I also revised the message.